### DISCUSSION TITLE
Apply consistent-type-specifier only for detecting type or interface

### DISCUSSION THREAD

**Discussion by aron-skaleet** [typescript-eslint/typescript-eslint#10281] [Technical Discussions]
### Body

Hello guys,
I have observed that consistent-type-specifier declare my class component as a type in the import declaration:

```
import type { Client } from "../shared/domains/client";
```

I expected to have:
```
import { Client } from "../shared/domains/client";
```

I think that when we declare: 
import type AType from "./Abc"
we are sure that AType is a type or interface exported declaration.

I feel weird to see:
"import type AClass from "./Abc"
and see that AClass is a class component instead of a type or an interface.

That is the reason I am asking if there is an option that we can add in order to use consistent-type-specifier and only apply the changes for a type or an interface exported component.

Thanks in advance for your response.

### Additional Info

Any additional info...

### Before you submit your discussion, please confirm the following. If any of these required steps are not taken, we may not be able to review your RFC. Help us to help you!

- [X] I have [read the discussions guidelines](https://typescript-eslint.io/contributing/discussions) and per those guidelines, this fits a discussion category, _not_ a help request or standard issue.
- [X] I have [searched for related discussions](https://github.com/typescript-eslint/typescript-eslint/discussions) and [searched for related issues](https://github.com/typescript-eslint/typescript-eslint/issues) and found none that match my proposal.
- [X] I have [read the FAQ](https://typescript-eslint.io/linting/troubleshooting) and my problem is not listed.

**Comment by JoshuaKGoldberg** [comment]
> I have observed that consistent-type-specifier

There's no rule named `consistent-type-specifier` in typescript-eslint. I'm going to assume you're asking about [`consistent-type-imports`](https://typescript-eslint.io/rules/consistent-type-imports). 

> That is the reason I am asking if there is an option that we can add

It looks like you're going for one or both of either:

* You're asking whether there is _currently_ an option to do what you want
* You're asking whether the project would be interested in adding an option if not

For the former, [`consistent-type-imports`](https://typescript-eslint.io/rules/consistent-type-imports) lists the available options. You can read through that to see the available options.

For the latter, you can [file a new issue: ✨ New Rule Option or Additional Check](https://github.com/typescript-eslint/typescript-eslint/issues/new?assignees=&labels=enhancement%3A+plugin+rule+option%2Cpackage%3A+eslint-plugin%2Ctriage&projects=&template=02-enhancement-rule-option.yaml&title=Enhancement%3A+%5Brule-name%5D+%3Ca+short+description+of+my+proposal%3E). Be sure to search for existing issues to make sure you're not filing a duplicate.

> -[x] I have [read the discussions guidelines](https://typescript-eslint.io/contributing/discussions) and per those guidelines, this fits a discussion category, _not_ a help request or standard issue.

This discussion was filed under the _"**Technical Discussions**: Deeper questions about typescript-eslint's architecture and/or APIs"_ category... but this is not a deep question about architecture or APIs. Request: please read the contributing guidelines more deeply before posting. They're there for a reason - to help you work with the systems we've set up for good reason. 🙂 

Closing this discussion as non-actionable. Thanks for filing!
